### PR TITLE
Hughsie/darwin

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: data/meson.build [hunk 1]
```diff
@@ -142,7 +142,7 @@ if build_daemon
     )
   endif
 
-  if launchctl.found()
+  if get_option('launchd').enabled() and launchctl.found()
```
Comment (superm1) [inline data/meson.build:145]: Is enabled() right? I would think allowed()

Comment (hughsie) [reply data/meson.build]: Fixed.

File: libfwupd/fwupd-common.c [hunk 2]
```diff
@@ -411,17 +419,21 @@ fwupd_build_machine_id(const gchar *salt, GError **error)
...
-			if (g_file_test(fns[i], G_FILE_TEST_EXISTS)) {
-				fn = fns[i];
+		g_ptr_array_add(fns, g_build_filename(FWUPD_SYSCONFDIR, "machine-id", NULL));
+		g_ptr_array_add(
+		    fns,
+		    g_build_filename(FWUPD_LOCALSTATEDIR, "lib", "dbus", "machine-id", NULL));
+		g_ptr_array_add(fns, g_strdup("/etc/machine-id"));
+		g_ptr_array_add(fns, g_strdup("/var/lib/dbus/machine-id"));
+		g_ptr_array_add(fns, g_strdup("/var/db/dbus/machine-id"));
+		g_ptr_array_add(fns, g_strdup("/usr/local/var/lib/dbus/machine-id"));
```
Comment (superm1) [inline libfwupd/fwupd-common.c:432]: I don't think it should be a hardcoded path should it?

Shouldn't this come from config.h? Or a path lookup?

Comment (hughsie) [reply libfwupd/fwupd-common.c:432]: Nah, it's literally hardcoded -- the homebrew people don't support any other path.

Comment (superm1) [reply libfwupd/fwupd-common.c:432]: Got it. Can you leave a comment for future selfs about this?

**Review by superm1 — APPROVED** [review]

File: libfwupd/fwupd-common.c (hunk 2 above)
Comment (hughsie) [reply libfwupd/fwupd-common.c:433]: Good idea, done.

**Review by superm1 — APPROVED** [review]

**Pull request merged**